vtex-apps/structured-data
# Add configuration for taxes

**What problem is this solving?**

This PR is meant to add to the prices sent to google the taxes (that's for stores when the prices are initially without a tax and the tax is separate in commertialOffer (the fields we're interested in are Price, spotPrice and Tax)

**How should this be manually tested?**

I created a configuration in the app's settings: https://structured--dacris.myvtex.com/admin/apps/vtex.structured-data@0.6.2/setup/

By default the decimals value is set to 2 and pricesWithTax is set to false to not impact any store that's currently using this app. Keep in mind that even though in manifest.json I set a default value, when initially installing the app, if you don't go to the app's settings and press Save changes (even without changing the default values) they will return as null from graphQL, as a fix for that scenario I set them in constants within Product.js in case they return null

OK -> install Chrome's extension for Structured Data and then go to a product

https://structured--dacris.myvtex.com/masca-faciala-de-uz-medical-tip-i-50buc-cut/p?__bindingAddress=www.dacris.net/

Generate the product response using the plugin and the lowPrice/highPrice/price should be either with the tax added or without, depending on which settings you configured.

As for the decimals, if you set it to 3 for example and the number is 6.239999 it won't say 6.240 it will say 6.24 (just keep that in mind when testing), it does work however for numbers such as 6.23333 (output will be 6.233)

**Screenshots or example usage:**

For this product

![image](https://user-images.githubusercontent.com/71461884/124725653-b0e46f00-df15-11eb-82fe-492fcf192383.png)

With tax added (in this case it's 19% but the Tax value from commertialOffer is a float number so it's added to the price)

![image](https://user-images.githubusercontent.com/71461884/124725349-75e23b80-df15-11eb-8dc1-6c000caedb42.png)

Without tax:

![image](https://user-images.githubusercontent.com/71461884/124725600-a629da00-df15-11eb-888f-309429726e4b.png)
